dotnet/src/Connectors/Connectors.Onnx/PoolingMode.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.SemanticKernel.Connectors.Onnx;
+
+/// <summary>Pooling mode used for creating the final sentence embedding.</summary>
+public enum EmbeddingPoolingMode
```

Comment:
nit, setting our Enums to bytes reduces the allocation? Is there a benefit of using small fingerprint value types for enums?

```suggestion
public enum EmbeddingPoolingMode : byte
```

Reply:
We'd generally advise against that in public surface area. As an implementation detail, in certain circumstances it can be advantageous, but not in a case like this. You're only going to have one of these on an object that's created rarely and that's likely also representing megabytes of other state (i.e. the model), and making this a byte instead of an int _might_ save three bytes. I say "might" because there's a good chance it wouldn't save anything, anyway, given how padding on the types work. And making this a byte is part of the public contract (it'd be a breaking change to change it), and as a byte it limits how many entries there could be here to just 8.